### PR TITLE
Feature/flin 84 kafka source resturants

### PR DESCRIPTION
**Pull request by VegSja**
## Describe your changes
- Added new RestaurantView.java file for getting relevant tables from kafka, joining and altering these into single view and stores this in mongoDB.
- Added class RestaurantTableEnvironment which abstracts away the fetching of the kafka streams into tables and the operations done on these tables
- Fetched all the relevant tables for the restaurant view.
- Created a sample SQL query on the tables to check that everything works as expected. :heavy_check_mark: 
- Created a simple mongoDB serializer to check if it is possible to store these tables in MongoDB. :heavy_check_mark: 

Using this sort of implementation for getting data from kafka topics into sql-like syntaxes makes it quite easy to make SQL like statements to join and alter the tables to get them to the format we would need, making the aggregation easier in the future.

## Results of running this code
This is an example of how one mongoDB document looks after going through this pipeline 
**NOTE**: This is still a very basic aggregation I did to test if getting the topics from kafka was working FLIN-119 and FLIN-120 will address making the document and aggregation more complex

```json
{
  "id": "4",
  "name": "Sushi Sensation",
  "phone": "+39 320 12345678",
  "email": "sushi.sensation@yahoo.com",
  "cuisine_type": "japanese",
  "price_range": "expensive",
  "vat_code": "65439876",
  "restaurant_id": "4",
  "take_away": "true",
  "delivery": "true",
  "dine_in": "true",
  "parking_lots": "10",
  "accessible": "true",
  "children_area": "true",
  "children_food": "true",
  "restaurant_id0": "4",
  "street": "Via Firenze",
  "address_number": "4",
  "zip_code": "45678",
  "city": "Firenze",
  "province": "FI",
  "country": "Italy",
  "id0": "7",
  "restaurant_id1": "4",
  "customer_id": "2",
  "rating": "KA==",
  "review_comment": "null"
}
```

## Github issue this pull requests addresses
Closes issue FLIN-84

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] I have updated/created the corresponding documentation

### CODE REVIEW

**Comment by PolimiGiovanniArriciati** [comment]
"Commit"

**Comment by PolimiGiovanniArriciati** [comment]
LOL

**Comment by PolimiGiovanniArriciati** [comment]
How draft is this? Do you want me to review it?

**Comment by VegSja** [comment]
This is ready for review now :crossed_fingers:

**Comment by PolimiGiovanniArriciati** [comment]
Can you test if the output is updated when inserting a new restaurant/updating services?

**Comment by VegSja** [comment]
It does, just tested it

**Comment by PolimiGiovanniArriciati** [comment]
To me looks like it does not work, can you test this case and give me some feedback. Maybe you did not delete the postgres container and the data was already in place

**Comment by VegSja** [comment]
I can write up a markdown file on how I run it. I am guessing there is some difference in how we run it. 

I accidentally deleted all my docker volumes and containers so the data was for sure not there

**Comment by PolimiGiovanniArriciati** [comment]
go for it, some more documentation is always welcome.
Maybe overwrite my test or delete it anyway... let's not clog up the project with redundant documents

File: postgres_entrypoint/init.sql [hunk 1]
```diff
@@ -9,6 +9,7 @@ CREATE TABLE IF NOT EXISTS restaurant_info (
     price_range VARCHAR(10),
     vat_code INT
 );
+ALTER TABLE restaurant_info REPLICA IDENTITY FULL;
```
Comment (PolimiGiovanniArriciati) [inline postgres_entrypoint/init.sql:12]: What are whose changes 😮

Comment (VegSja) [reply postgres_entrypoint/init.sql:12]: It's for debezium:
https://stackoverflow.com/questions/59799503/postgres-debezium-does-not-publish-the-previous-state-of-a-record

**Review by PolimiGiovanniArriciati — APPROVED** [review]

**Pull request merged**